tree/master/cloud/src/solution/Microsoft.Legal.MatterCenter.Service/Microsoft.Legal.MatterCenter.Repository/Matter/MatterRepository.cs

```diff
@@ -1514,24 +1519,193 @@ internal string GetStampPropertyValue(Dictionary<string, object> stampedProperty
+                {
+                    foreach (IList<string> userNames in matter.AssignUserNames)
+                    {
+                        currentUsers += separator + string.Join(ServiceConstants.SEMICOLON, userNames.Where(user => !string.IsNullOrWhiteSpace(user)));
+                        separator = ServiceConstants.DOLLAR + ServiceConstants.PIPE + ServiceConstants.DOLLAR;
+                    }
+                }
+                return currentUsers;
+            }catch(Exception ex)
+            {
+                customLogger.LogError(ex, MethodBase.GetCurrentMethod().DeclaringType.Name, MethodBase.GetCurrentMethod().Name, logTables.SPOLogTable);
+                throw;
+            }
+        }
+
+        public bool SetPermission(ClientContext clientContext, IList<IList<string>> assignUserName, IList<string> permissions, string listName) => spList.SetPermission(clientContext, assignUserName, permissions, listName);
+
+        public GenericResponseVM ShareMatterToExternalUser(MatterInformationVM matterInformation) => extrnalSharing.ShareMatter(matterInformation);
+
+        public bool OneNoteUrlExists(MatterInformationVM matterInformation) => search.PageExists(matterInformation.Client, matterInformation.RequestedUrl);
+
+        /// <summary>
+        /// This method will get all the Field columns of the content type as the JSON object which is used to render the dynamic UI.
+        /// </summary>
+        /// <param name="contentTypeName"></param>
+        /// <param name="client"></param>
+        /// <returns>Json</returns>
+        public string GetMatterProvisionExtraProperties(string contentTypeName, Client client)
+        {
+            try
+            {
+                var clientContext = spoAuthorization.GetClientContext(client.Url);
+                FieldCollection fieldCollection = contentTypeName.GetFieldsInContentType(clientContext);
+
+                //To get configuration settings...
+                var listQuery = string.Format(CultureInfo.InvariantCulture, camlQueries.MatterConfigurationsListQuery,
+                        searchSettings.ManagedPropertyTitle, searchSettings.MatterConfigurationTitleValue);
+                ListItem settingsItem = spList.GetData(clientContext,
+                    listNames.MatterConfigurationsList, listQuery).FirstOrDefault();
+
+                Newtonsoft.Json.Linq.JObject settingConfig = (Newtonsoft.Json.Linq.JObject)
+                    JsonConvert.DeserializeObject(WebUtility.HtmlDecode(Convert.ToString(
+                        settingsItem.FieldValues["ConfigurationValue"])));
+
+                IList<MatterExtraFields> AddFields = new List<MatterExtraFields>();
+
+                if (settingConfig != null && settingConfig.GetValue("AdditionalFieldValues") != null)
+                {
+                    foreach (var objField in settingConfig.GetValue("AdditionalFieldValues"))
+                    {
+                        AddFields.Add(new MatterExtraFields

```

Comment:
Logic in foreach loop is to retrieve  data if available which are set for MatterExtraProperties from Settings page for client for the sub area of law (term like copyright). if yes then we will read the settings of the site columns metadata like isdisplayin ui and ismandatory and display accordigly.